### PR TITLE
Makes exported function code more user-friendly to read

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR refactors how we capture and return exported functions to the user.
The code is captured when a workflow is created by adding a special read-only file called `source.py`. This file isn't used as part of any operator execution, but it is just there as a user-friendly copy of the code that has been serialized into `model.pkl`.

## Related issue number (if any)
ENG 1190

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

UPDATE:
The files now look like
Screen Shot 2022-08-29 at 4 48 40 PM

Demo: https://www.loom.com/share/a553c71f4d9b4216a7c2b22db30073ad

### CODE REVIEW

File: sdk/aqueduct/utils.py [hunk 1]
```diff
@@ -110,9 +110,11 @@ def retention_policy_from_latest_runs(k_latest_runs: int) -> RetentionPolicy:
 
 MODEL_FILE_NAME = "model.py"
 MODEL_PICKLE_FILE_NAME = "model.pkl"
+FUNCTION_SOURCE_FILE_NAME = "source.py"
```
Comment (saurav-c) [inline sdk/aqueduct/utils.py:113]: @vsreekanti Is it too restrictive to make `source.py` a reserved file?

Comment (vsreekanti) [reply sdk/aqueduct/utils.py:113]: I would prefer not to. Would it be possible to make the filename in the zipfile the name of the function?

Comment (saurav-c) [reply sdk/aqueduct/utils.py:113]: Sure, we'll just have to prevent users from uploading file dependencies with the same name as the function.

Comment (vsreekanti) [reply sdk/aqueduct/utils.py:113]: Yep, I think that's fine.

Comment (saurav-c) [reply sdk/aqueduct/utils.py:113]: Fixed, the files now look like:
Screen Shot 2022-08-29 at 4 48 40 PM

Comment (vsreekanti) [reply sdk/aqueduct/utils.py]: Yep, that's perfect.

File: sdk/aqueduct/utils.py [hunk 2]
```diff
@@ -197,6 +199,13 @@ def serialize_function(
         with open(os.path.join(dir_path, MODEL_PICKLE_FILE_NAME), "wb") as f:
             pickle.dump(func, f)
 
+        # Write function source code to file
+        source_file = "{}.py".format(op_name)
+        with open(os.path.join(dir_path, source_file), "w") as f:
+            source = inspect.getsource(func)
+            print(source)
```
Comment (kenxu95) [inline sdk/aqueduct/utils.py:206]: is this print supposed to be here? If it is, was should use the logger.

File: src/golang/cmd/server/handler/export_function.go [hunk 3]
```diff
@@ -125,13 +135,92 @@ func (h *ExportFunctionHandler) Perform(ctx context.Context, interfaceArgs inter
...
+		// than the current directory name that is a unique UUID
+		parts[0] = operatorName
+		zipFileName := strings.Join(parts, "/")
+
+		if err := writeZipFile(zipWriter, zipFile, zipFileName); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := zipWriter.Close(); err != nil {
```
Comment (kenxu95) [inline src/golang/cmd/server/handler/export_function.go:197]: I'm not familiar with the zipWriter interface, is there a reason we don't `defer` this after the zipWriter creation?

File: sdk/aqueduct/utils.py [hunk 4]
```diff
@@ -197,6 +199,13 @@ def serialize_function(
         with open(os.path.join(dir_path, MODEL_PICKLE_FILE_NAME), "wb") as f:
             pickle.dump(func, f)
 
+        # Write function source code to file
+        source_file = "{}.py".format(op_name)
+        with open(os.path.join(dir_path, source_file), "w") as f:
+            source = inspect.getsource(func)
```
Comment (kenxu95) [inline sdk/aqueduct/utils.py:205]: Based on the loom vid, this won't capture the import statements right? Should that be a nice-to-have follow up (is it even possible)

**Review by kenxu95 — APPROVED** [review]
nice!

File: src/golang/cmd/server/handler/export_function.go (hunk 3 above)
Comment (saurav-c) [reply src/golang/cmd/server/handler/export_function.go:212]: We have to explicitly close the `zipWriter` before we can read its content: https://stackoverflow.com/a/55623815

File: sdk/aqueduct/utils.py (hunk 2 above)
Comment (saurav-c) [reply sdk/aqueduct/utils.py:206]: Removed

File: sdk/aqueduct/utils.py (hunk 4 above)
Comment (saurav-c) [reply sdk/aqueduct/utils.py:205]: Yup that's right. I don't think it's easily possible and not sure if its worth the engineering effort cc @vsreekanti

**Pull request merged**